deb/debian/preinst

```diff
@@ -0,0 +1,53 @@
+#!/bin/sh
+# preinst script for stackdriver-agent
+#
+# see: dh_installdeb(1)
+
+set -e
+
+# summary of how this script can be called:
+#        * <preinst> `install'
+#        * <preinst> `install' <old-version> <new-version>
+#        * <new-preinst> `upgrade' <old-version> <new-version>
+#        * <old-preinst> `abort-upgrade' <old-version>
+# for details, see http://www.debian.org/doc/debian-policy/ or
+# the debian-policy package
+
+. /usr/share/debconf/confmodule
+
+OPT_COLLECTD="/opt/stackdriver/collectd/etc"
+ARCHIVE_DIR="/etc/stackdriver/archive"
+
+case "$1" in
+    install)
+        ;;
+
+    upgrade)
+      if [ ! -d "$ARCHIVE_DIR" ]; then
+        mkdir -p "$ARCHIVE_DIR"
+      fi
+
+      if [ -f "$OPT_COLLECTD/collectd.conf.tmpl" ]; then
+        cp "$OPT_COLLECTD/collectd.conf.tmpl" \
+          "$ARCHIVE_DIR/collectd.conf.tmpl.old"
+      fi
+        ;;
+
+    abort-upgrade)
+      if [ -f "$OPT_COLLECTD/collectd.conf.tmpl" && -d "$ARCHIVE_DIR" ]; then
```

Comment:
Why delete the archive directory _only_ if the tmpl file exists? Is there any reason not to always delete it?

(More generally: should 'abort-upgrade' undo what was done in 'upgrade'?)

Reply:
I think upgrade and abort-upgrade are now aligned. WDYT? 

The original idea behind this is forward compatibility. Lets say our broken version is 370 and the user first upgrades to a version 380 which has this fix. We move `collectd.conf.tmpl`. However, if the user were to upgrade from 380 --> 390 and abort-upgrade is called, we don't want to blow away the `$ARCHIVE_DIR` because it was created as part of a different upgrade and doesn't undo what we did in the upgrade step for this particular upgrade (i.e. 380 --> 390)

Reply:
Got it. Makes sense. Thanks.